pkg/scalers/externalscaler/externalscaler.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+package externalscaler;
+
+import "google/protobuf/empty.proto";
+
+service ExternalScaler {
+    rpc New(NewRequest) returns (google.protobuf.Empty) {}
+    rpc IsActive(ScaledObjectRef) returns (IsActiveResponse) {}
+    rpc GetMetricSpec(ScaledObjectRef) returns (GetMetricSpecResponse) {}
+    rpc GetMetrics(GetMetricsRequest) returns (GetMetricsResponse) {}
+    rpc Close(ScaledObjectRef) returns (google.protobuf.Empty) {}
+}
+
+message ScaledObjectRef {
+    string name = 1;
+    string namespace = 2;
+}
+
+message NewRequest {
```

Comment:
shouldn't this return an error if the metadata parse fails?

Reply:
All gRPC interfaces return an error by default. You don't need to explicitly mention that in the protobuf. [Here is a sample server implementing the method] (https://github.com/patnaikshekhar/KEDA-External-Scaler-Example/blob/master/main.go#L66)